### DISCUSSION TITLE
Question - Write bandwidth is much higher than Read bandwidth?

### DISCUSSION THREAD

**Discussion by kuhyungc** [NVIDIA/gdrcopy#201] [Q&A]
Hello,

This is Ku, Developer Relations Manager, NVIDIA. One of my customers is testing this gdrcopy and asked a question, "How can we explain the reason why Write bandwidth is much higher than Read bandwidth?" Can you shed a light on mechanism how Write B/W and Read B/W are calculated and why W's is higher than R's?

$ GDRCOPY_ENABLE_LOGGING=1 GDRCOPY_LOG_LEVEL=0 LD_LIBRARY_PATH=$PWD:$LD_LIBRARY_PATH numactl -N 0 -l copybw -d 0 -s $((64 * 1024)) -o $((0 * 1024)) -c $((64 * 1024))
GPU id:0 name:Tesla K40m PCI domain: 0 bus: 2 device: 0
GPU id:1 name:Tesla K80 PCI domain: 0 bus: 132 device: 0
GPU id:2 name:Tesla K80 PCI domain: 0 bus: 133 device: 0
selecting device 0
testing size: 65536
rounded size: 65536
device ptr: 2305ba0000
bar_ptr: 0x7fe60956c000
info.va: 2305ba0000
info.mapped_size: 65536
info.page_size: 65536
page offset: 0
user-space pointer:0x7fe60956c000
BAR writing test, size=65536 offset=0 num_iters=10000
DBG:  sse4_1=1 avx=1 sse=1 sse2=1
DBG:  using AVX implementation of gdr_copy_to_bar
**BAR1 write BW: 9793.23MB/s**
BAR reading test, size=65536 offset=0 num_iters=100
DBG:  using SSE4_1 implementation of gdr_copy_from_bar
**BAR1 read BW: 787.957MB/s**
unmapping buffer
unpinning buffer
closing gdrdrv

**Comment by pakmarkthub** [comment]
Already discussed offline.
